src/ObjectPool/src/DefaultObjectPool.cs

```diff
@@ -8,6 +8,10 @@
 
 namespace Microsoft.Extensions.ObjectPool
 {
+    /// <summary>
+    /// Default implementation of <see cref="ObjectPool{T}"/>.
+    /// </summary>
+    /// <typeparam name="T">The type to pool objects for.</typeparam>
```

Comment:
We should capture the detail that this implementation keeps a cache of *retained objects*. This means that if you check out twenty objects, and return twenty objects then `maximumRetained` will be kept by the pool, and `20 - maximumRetained` will be available to be GCed.